File: PingSysInventory/PingSysInventory.Web/Modules/ServerTypes/Inventory/StockMovementsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { StockMovementsRow } from "./StockMovementsRow";

export interface StockMovementsColumns {
    MovementId: Column<StockMovementsRow>;
    ProductName: Column<StockMovementsRow>;
    MovementType: Column<StockMovementsRow>;
    Quantity: Column<StockMovementsRow>;
    MovementDate: Column<StockMovementsRow>;
    Notes: Column<StockMovementsRow>;
    InsertUserId: Column<StockMovementsRow>;
}

export class StockMovementsColumns extends ColumnsBase<StockMovementsRow> {
    static readonly columnsKey = 'Inventory.StockMovements';
    static readonly Fields = fieldsProxy<StockMovementsColumns>();
}

File: PingSysInventory/PingSysInventory.Web/Modules/Inventory/Forms/StockMovementsColumns.cs
using PingSysInventory.Inventory;

namespace PingSysInventory.Inventory.Forms;

[ColumnsScript("Inventory.StockMovements")]
[BasedOnRow(typeof(StockMovementsRow), CheckNames = true)]
public class StockMovementsColumns
{
    [DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MovementId { get; set; }

    [Width(220)]
    public string ProductName { get; set; }

    [Width(100)]
    public int MovementType { get; set; }

    [Width(80), AlignRight]
    public int Quantity { get; set; }

    [Width(140)]
    public DateTime MovementDate { get; set; }

    [Width(200)]
    public string Notes { get; set; }

    [Width(90), AlignRight]
    public int InsertUserId { get; set; }
}
